plugins/scanners/fossid/src/main/kotlin/FossIdConfig.kt

```diff
@@ -19,232 +19,113 @@
- *
- * The example
- *
- * `mapExampleRepo = https://my-repo.example.org(?<repoPath>.*) -> ssh://my-mapped-repo.example.org${repoPath}`
- *
- * would change the scheme from "https" to "ssh" and the host name for all repositories hosted on
- * "my-repo.example.org". With
- *
- * `mapAddCredentials =
- *   (?<scheme>)://(?<host>)(?<port>:\\d+)?(?<repoPath>.*) -> ${scheme}://#user:#password@${host}${port}${repoPath}`
- *
- * every repository URL would be added credentials. Mappings are applied in the order they are defined.
- */
+/** A data class that holds the configuration options supported by the [FossId] scanner. */
 data class FossIdConfig(
     /** The URL where the FossID service is running. */
     val serverUrl: String,
 
     /** The user to authenticate against the server. */
-    val user: String,
+    val user: Secret,
 
     /** The API key to access the FossID server. */
-    val apiKey: String,
+    val apiKey: Secret,
 
     /** The name of the FossID project. If `null`, the name will be determined from the repository URL. */
     val projectName: String?,
 
     /** The pattern for scan names when scans are created on the FossID instance. If null, a default pattern is used. */
     val namingScanPattern: String?,
 
-    /** Flag whether the scanner should wait for the completion of FossID scans. */
+    /**
+     * When set to false, ORT does not wait for repositories to be downloaded nor scans to be completed. As a
+     * consequence, scan results won't be available in the ORT result.
+     */
+    @OrtPluginOption(defaultValue = "true")
     val waitForResult: Boolean,
 
     /** Flag whether failed scans should be kept. */
+    @OrtPluginOption(defaultValue = "false")
     val keepFailedScans: Boolean,
 
-    /** Flag whether delta scans should be triggered. */
+    /** If set, ORT will create delta scans. When only changes in a repository need to be scanned, delta scans reuse the
+     *  identifications of the latest scan on this repository to reduce the number of findings. If *deltaScans* is set
+     *  and no scan exists yet, an initial scan called "origin" scan will be created.
+     */
+    @OrtPluginOption(defaultValue = "false")
     val deltaScans: Boolean,
 
-    /** A maximum number of delta scans to keep for a single repository. */
+    /**
+     * This setting can be used to limit the number of delta scans to keep for a given repository. So if another delta
+     * scan is created, older delta scans are deleted until this number is reached. If unspecified, no limit is enforced
+     * on the number of delta scans to keep. This property is evaluated only if delta scans are enabled.
+     */
+    @OrtPluginOption(defaultValue = Int.MAX_VALUE.toString())
     val deltaScanLimit: Int,
 
     /**
      * Configure to automatically detect license declarations. Uses the `auto_identification_detect_copyright` setting.
      */
+    @OrtPluginOption(defaultValue = "false")
```

Comment:
Is a default value needed here and underneath ?
A non-nullable boolean should be automatically be initialized to `false` no ?

Reply:
It is, the plugin API requires to be explicit about default values, the only exception are nullable properties.